rasa/core/policies/ted_policy.py

```diff
@@ -925,6 +925,11 @@ def _prepare_layers(self) -> None:
             self.config[TRANSFORMER_SIZE][DIALOGUE],
             self.config[DROP_RATE_DIALOGUE],
             self.config[DROP_RATE_ATTENTION],
+            # use bidirectional transformer, because
+            # we will invert dialogue sequence so that the last turn is located
+            # at the first position and would always have
+            # exactly the same positional encoding
+            unidirectional=not self.use_only_last_dialogue_turns,
```

Comment:
I'd say lets use a variable explicitly to denote when the dialogue transformer is supposed to be bi-directional. Right now, on first look it seems a bit blurry on what `config[UNIDIRECTIONAL]`, `self.use_only_last_dialogue_turns` are intended to do. Specially, now that we are using `config[UNIDIRECTIONAL]` for feature sequence layer transformers as well.

Reply:
basically `config[UNIDIRECTIONAL]` is a parameter from `DIET` and is only for text transformers

Reply:
> I'd say lets use a variable explicitly

could you please elaborate, what do you mean?

Reply:
How about we create an instance variable in `TED` named `dialogue_bidirectional`(or something similar) which is set in the constructor itself based on `self.use_only_last_dialogue_turns`. Then here and everywhere else where we make an operation because of transformer being bidirectional and input sequence being reversed, we explicitly use that variable instead of `use_only_last_dialogue_turns`. Right now, the usage of `use_only_last_dialogue_turns` is overloaded across a couple of different contexts.

Reply:
> first look it seems a bit blurry on what config[UNIDIRECTIONAL]
I updated the comment in `defaults`

Reply:
they have to be linked they are equivalent to `isinstance(self.featurizer, MaxHistoryTrackerFeaturizer)` if one is `True` the other one should be `False` the other combination is not allowed. I can rename `use_only_last_dialogue_turns` into `max_history_featurizer_is_used`

Reply:
That would be better I guess, since both are derived from the same information,  it will be more explicit on how they are computed.

Reply:
renamed